src/lib/languages/mod.rs

```diff
@@ -0,0 +1,32 @@
+mod javascript;
+
+use super::*;
+pub use javascript::JavaScript;
+
+#[derive(Debug)]
+pub struct LangError {
+  language: String
+}
+
+pub enum Lang {
+  JS
+}
+
+impl Into<&'static str> for Lang {
+  fn into(self) -> &'static str {
+    match self {
+      Self::JS => "JavaScript"
+    }
+  }
+}
+
+
+pub fn generate(parser: Parser, lang: Lang) -> Result<String, String> {
```

Comment:
I would prefer to have this return a enum error type like `ParsingErrorType` because stage 3 will have a lot of errors that are equal.  
I think you can best add a copy of the `ParsingErrorType` logic and add the new error you just created to `ParsingErrorType`.